TTS/trainer.py

```diff
@@ -640,10 +651,24 @@ def _optimize(
                 update_lr_scheduler = scale_prev <= scaler.get_scale()
                 loss_dict["amp_scaler"] = scaler.get_scale()  # for logging
         else:
+            if self.model.capacitron_vae:
+                loss_dict["capacitron_vae_beta_loss"].backward()
+                capacitron_SGD_optimizer.step()
+                capacitron_SGD_optimizer.zero_grad()
+                optimizer.zero_grad()
+                # Gradient clipping below needs to exclude capacitron_beta
+                model_params_to_clip = []
+                for name, param in model.named_parameters():
+                    if param.requires_grad:
+                        if name != "capacitron_vae_layer.beta":
+                            model_params_to_clip.append(param)
+            else:
+                model_params_to_clip = model.parameters()
+
             # main model optimizer step
             loss_dict["loss"].backward()
             if grad_clip > 0:
-                grad_norm = torch.nn.utils.clip_grad_norm_(model.parameters(), grad_clip)
+                grad_norm = torch.nn.utils.clip_grad_norm_(model_params_to_clip, grad_clip)
```

Comment:
Gradient clipping must exclude `beta`

Reply:
sry but what is `beta` ?

Reply:
`beta` is the single lagrange-multiplier like constant for the VAE part of the loss function that is optimized by a separate optimizer (SGD). It needs to be excluded from the parameters for the main optimizer.

Reply:
and how about if you disable grad_clip for the whole model?

Reply:
need gradient clipping according to the paper (haven't tested without it yet)

Reply:
if it works without clipping after we solve the rest we can take a look at it again. Probably in the paper, they used it since it is the default on what they use.